src/components/Search/search.js

```diff
@@ -1,14 +1,36 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './search.css';
+import { useHistory } from 'react-router-dom';
 
-const Search = () => (
-  <div className="search">
-    <h3 className="SearchHeader">Find your movie</h3>
-    <div className="search-bar">
-      <input className="search-input" type="text" />
-      <button className="search-button" type="button">Search</button>
+const Search = () => {
+  const [searchInput, setSearchInput] = useState('');
+  
+  const history = useHistory();
+  
+  function handleClick() {
+    if (searchInput !== '') {
+      history.push(`/?search=${searchInput}`);
+    }
+  }
+  
+  function enterPressed(event) {
+    let code = event.keyCode || event.which;
+    if (code === 13) { //13 is the enter keycode
+      handleClick();
+    }
+  }
+  
+  return (
+    <div className="search">
+      <h3 className="SearchHeader">Find your movie</h3>
+      <div className="search-bar">
+        <input className="search-input" type="text"
+               onKeyPress={enterPressed.bind(this)}
```

Comment:
Why are you binding the function? Arrow function has no own context preserving the parent one. So basically you would bind the function to the module context. Still you ain't need it

Reply:
fixed